snapcraft/parts/plugins/flutter.py

```diff
@@ -0,0 +1,63 @@
+from typing import List, Dict, Set, cast, Literal, Any
+
+from craft_parts import plugins
+
+
+class FlutterPluginProperties(plugins.PluginProperties, plugins.PluginModel):
+    flutter_branch: Literal["stable", "master", "dev"] = "stable"
+    flutter_target: str = "lib/main.dart"
+
+    @classmethod
+    def unmarshal(cls, data: Dict[str, Any]) -> "CondaPluginProperties":
+        """Populate class attributes from the part specification.
+
+        :param data: A dictionary containing part properties.
+
+        :return: The populated plugin properties data object.
+
+        :raise pydantic.ValidationError: If validation fails.
+        """
+        plugin_data = plugins.extract_plugin_properties(
+            data,
+            plugin_name="flutter",
+        )
+        return cls(**plugin_data)
+
+
+class FlutterPlugin(plugins.Plugin):
+
+    properties_class = FlutterPluginProperties
+
+    def get_build_snaps(self) -> Set[str]:
+        return {}
+
+    def get_build_packages(self) -> Set[str]:
+        return {
+            "clang",
+            "git",
+            "cmake",
+            "ninja-build",
+            "unzip",
```

Comment:
the expectation is that the gnome extension makes use of the gnome sdk snap which in itself provides gtk3.
What will happen in the close future in this code base, is to add the plugin validation mechanism so that a developer can provide the dependencies as they see fit (cmake deb or snap, etc)

Reply:
Is the same true for `xz-utils`? I don't see it mentioned here https://github.com/ubuntu/gnome-sdk/blob/gnome-3-38-2004-sdk/snapcraft.yaml .

Reply:
it's part of build-essential, so already there
```
sergiusens@imladris:~/repo/snapcore/snapcraft$ lxc start --project snapcraft snapcraft-myapp-on-amd64-for-amd64-58461699 
sergiusens@imladris:~/repo/snapcore/snapcraft$ lxc exec --project snapcraft snapcraft-myapp-on-amd64-for-amd64-58461699 -- bash
snapcraft-myapp-on-amd64-for-amd64-58461699 # dpkg -l xz-utils
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version        Architecture Description
+++-==============-==============-============-=================================
ii  xz-utils       5.2.5-2ubuntu1 amd64        XZ-format compression utilities
```